File: frontend/src/app/core/models/proceso.ts
export interface Proceso {
  // Información de la Entidad
  entidad: string;
  nit_entidad: string;
  departamento_entidad: string;
  ciudad_entidad: string;
  ordenentidad: string;
  codigo_entidad: number | null;

  // Información del Proceso
  id_del_proceso: string;
  referencia_del_proceso: string;
  nombre_del_procedimiento: string;
  descripci_n_del_procedimiento: string;
  fase: string;
  estado_resumen: string;
  estado_del_procedimiento: string;
  id_estado_del_procedimiento: number | null;

  // Contratación
  modalidad_de_contratacion: string;
  justificaci_n_modalidad_de: string;
  tipo_de_contrato: string;
  subtipo_de_contrato: string;
  duracion: number | null;
  unidad_de_duracion: string;

  // Fechas
  fecha_de_publicacion_del: string;
  fecha_de_ultima_publicaci: string;
  fecha_de_recepcion_de: string;
  fecha_de_apertura_de_respuesta: string;
  fecha_de_apertura_efectiva: string;
  fecha_adjudicacion: string;

  // Información Económica
  precio_base: number | null;
  adjudicado: string;
  valor_total_adjudicacion: number | null;

  // Proveedor Adjudicado
  codigoproveedor: string;
  nombre_del_proveedor: string;
  nit_del_proveedor_adjudicado: string;
  departamento_proveedor: string;
  ciudad_proveedor: string;

  // Estadísticas
  proveedores_invitados: number | null;
  proveedores_con_invitacion: number | null;
  proveedores_que_manifestaron: number | null;
  respuestas_al_procedimiento: number | null;
  respuestas_externas: number | null;
  conteo_de_respuestas_a_ofertas: number | null;
  proveedores_unicos_con: number | null;
  visualizaciones_del: number | null;
  numero_de_lotes: number | null;

  // Clasificación
  codigo_principal_de_categoria: string;
  categorias_adicionales: string;

  // Enlace
  urlproceso: string;
}

File: backend/app/models/proceso.py
from pydantic import BaseModel


class Proceso(BaseModel):
    # ==========================
    # Información de la Entidad
    # ==========================

    entidad: str = ""
    nit_entidad: str = ""
    departamento_entidad: str = ""
    ciudad_entidad: str = ""
    ordenentidad: str = ""
    codigo_entidad: int | None = None

    # ==========================
    # Información del Proceso
    # ==========================

    id_del_proceso: str = ""
    referencia_del_proceso: str = ""
    nombre_del_procedimiento: str = ""
    descripci_n_del_procedimiento: str = ""
    fase: str = ""

    estado_resumen: str = ""
    estado_del_procedimiento: str = ""
    id_estado_del_procedimiento: int | None = None

    # ==========================
    # Contratación
    # ==========================

    modalidad_de_contratacion: str = ""
    justificaci_n_modalidad_de: str = ""

    tipo_de_contrato: str = ""
    subtipo_de_contrato: str = ""

    duracion: float | None = None
    unidad_de_duracion: str = ""

    # ==========================
    # Fechas
    # ==========================

    fecha_de_publicacion_del: str = ""
    fecha_de_ultima_publicaci: str = ""

    fecha_de_recepcion_de: str = ""
    fecha_de_apertura_de_respuesta: str = ""
    fecha_de_apertura_efectiva: str = ""

    fecha_adjudicacion: str = ""

    # ==========================
    # Información Económica
    # ==========================

    precio_base: float | None = None

    adjudicado: str = ""
    valor_total_adjudicacion: float | None = None

    # ==========================
    # Proveedor Adjudicado
    # ==========================

    codigoproveedor: str = ""
    nombre_del_proveedor: str = ""
    nit_del_proveedor_adjudicado: str = ""

    departamento_proveedor: str = ""
    ciudad_proveedor: str = ""

    # ==========================
    # Estadísticas
    # ==========================

    proveedores_invitados: int | None = None
    proveedores_con_invitacion: int | None = None
    proveedores_que_manifestaron: int | None = None

    respuestas_al_procedimiento: int | None = None
    respuestas_externas: int | None = None

    conteo_de_respuestas_a_ofertas: int | None = None
    proveedores_unicos_con: int | None = None

    visualizaciones_del: int | None = None
    numero_de_lotes: int | None = None

    # ==========================
    # Clasificación
    # ==========================

    codigo_principal_de_categoria: str = ""
    categorias_adicionales: str = ""

    # ==========================
    # Enlace
    # ==========================

    urlproceso: str = ""
